vllm/v1/engine/async_llm.py

```diff
@@ -0,0 +1,317 @@
+    ) -> AsyncGenerator[RequestOutput, None]:
+
+        # We start the output_handler on the first call to generate() so that
+        # we can call __init__ before the event loop starts, which enables us
+        # to handle startup failure gracefully in the OpenAI server.
+        if not self.is_output_handler_running:
+            self.output_handler = asyncio.create_task(
+                self._run_output_handler())
+            self.is_output_handler_running = True
+
+        async for output in await self.add_request(
+                request_id,
+                prompt,
+                sampling_params,
+                lora_request=lora_request,
+                trace_headers=trace_headers,
+                prompt_adapter_request=prompt_adapter_request,
+                priority=priority,
+        ):
+            yield output
+
+    async def _run_output_handler(self):
+
+        # TODO: add weakref from current AsyncLLMEngine
+        # TODO: shutdown remote worker execution loop (once TP enabled)
+
+        logger.debug("Starting output handler busy loop in background loop.")
+
+        try:
+            while True:
+                # Get EngineCoreOutput from the EngineCore.
+                outputs = await self.engine_core.get_output_async()
+
+                # Detokenize based on the output.
+                request_outputs, reqs_to_abort = self.detokenizer.step(outputs)
+
+                # Put the RequestOutputs into the per-request AsyncStream.
+                # NOTE(rob): we could do the streaming in the detokenizer.
+                self._send_to_streams(request_outputs)
+
+                # Abort any requests that finished due to stop strings.
+                await self.abort_requests(reqs_to_abort)
+
+        except BaseException as e:
+            logger.error(e)
+            raise e
+
+    # TODO: can we eliminate these?
+
+    async def abort(self, request_id: str) -> None:
+        # Note: Who Calls this?
+        raise ValueError("Not Supported on V1 yet.")
+
+    def encode(
+        self,
+        prompt: PromptType,
+        pooling_params: PoolingParams,
+        request_id: str,
+        lora_request: Optional[LoRARequest] = None,
+        trace_headers: Optional[Mapping[str, str]] = None,
+        priority: int = 0,
+    ):
+        raise ValueError("Not Supported on V1 yet.")
+
+    async def get_model_config(self) -> ModelConfig:
+        return self.model_config
+
+    async def get_decoding_config(self):
+        raise ValueError("Not Supported on V1 yet.")
+
+    async def get_tokenizer(
+        self,
+        lora_request: Optional[LoRARequest] = None,
+    ) -> AnyTokenizer:
+        assert lora_request is None
+        return self.detokenizer.tokenizer
+
+    async def is_tracing_enabled(self) -> bool:
+        return False
+
+    async def do_log_stats(
+        self,
+        scheduler_outputs=None,
+        model_output=None,
+    ) -> None:
+        logger.debug("Called do_log_stats.")
```

Comment:
This we should definitely eliminate imo

Reply:
These will be eliminated in the end. For now, we need to conform to `EngineClient` so we dont have to change the server API